### PR TITLE
PROPULSION: Implement thruster mapper

### PR DESCRIPTION
**Pull request by jpanikulam**
Thruster mapper implementation

Some notes:
- Not sure if this belongs in dev. The node is functional, except for two fairly minor features.
- Snuck in some necessary fixes to utils to make these commits work (How litigious should we be on that?)
- Includes a simple rostest

### CODE REVIEW

File: gnc/sub8_thruster_mapper/nodes/mapper.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        '''
+        self.min_thrust, self.max_thrust = self.get_ranges()
+        # Make thruster layout
+        self.B = self.generate_B(layout)
+        self.wrench_sub = rospy.Subscriber('/wrench', WrenchStamped, self.request_wrench_cb, queue_size=1)
+        self.thruster_pub = rospy.Publisher('/thrust', Thrust, queue_size=1)
+
+    @thread_lock(lock)
+    def update_layout(self, srv):
+        '''Respond to a thruster-out alarm'''
```
Comment (pemami4911) [inline gnc/sub8_thruster_mapper/nodes/mapper.py]: The name of this function and the docstring are a bit confusing. Is the sole purpose of this function for responding to a thruster-out alarm?

**Comment by pemami4911** [comment]
Looks good!

**Comment by jpanikulam** [comment]
Added a new commit fixing the docstrings in mapper.py as per your comment on the intial PR.

See: 1f7d22d437e35dd7c6daa4d4ef51433c506c8189

**Comment by jpanikulam** [comment]
pull?

**Comment by jpanikulam** [comment]
<3

**Pull request merged**